### PR TITLE
feat: [CI-5775]: Support debug using tmate

### PR DESCRIPTION
**Pull request by HenAmar**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/38914)

### CODE REVIEW

File: product/ci/addon/docker/Dockerfile.linux.arm64 [hunk 1]
```diff
@@ -23,6 +23,12 @@ COPY split_tests /usr/local/bin/
 RUN chmod +x /usr/local/bin/ci-addon-linux \
     && chmod +x /usr/local/bin/split_tests
 
+# Install tmate
+  wget https://github.com/tmate-io/tmate/releases/download/2.4.0/tmate-2.4.0-static-linux-amd64.tar.xz \
```
Comment (HenAmar) [inline product/ci/addon/docker/Dockerfile.linux.arm64:27]: add env var for version

**Review by archit-harness — COMMENTED** [review]
> Reviewed 2 of 7 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38914)* status: 21 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`pipeline-service/modules/pms-contracts/src/main/proto/io/harness/pms/contracts/plan/trigger_info.proto` line 17 at r3](https://reviewable.io/reviews/harness/harness-core/38914#-NH-V8k35DKw1rRMSKIt:-NHxzHxe1hWGt2_MZxX-:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/1bce8dc93fe93262764074dbe7331c5f8ea5ab42/pipeline-service/modules/pms-contracts/src/main/proto/io/harness/pms/contracts/plan/trigger_info.proto#L17)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

i think this property should be part of ExecutionMetadata
</blockquote></details>

done

**Review by archit-harness — APPROVED** [review]
> Reviewed 8 of 18 files at r7, 1 of 1 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38914)* status: 15 file(s) to review.

**Review by naidu-harness — APPROVED** [review]
> Reviewed 2 of 16 files at r2, 3 of 17 files at r3, 18 of 18 files at r7, 1 of 1 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38914)* status: :shipit: complete! ✓

**Pull request merged**